openshift/che-plugin-registry.yml

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (c) 2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+---
+apiVersion: v1
+kind: Template
+metadata:
+  name: che-plugin-registry-service
+objects:
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: che-plugin-registry
+    name: che-plugin-registry
+  spec:
+    replicas: 1
+    selector:
+      app: che-plugin-registry
+      deploymentconfig: che-plugin-registry
+    strategy:
+      rollingParams:
+        intervalSeconds: 1
+        maxSurge: 25%
+        maxUnavailable: 25%
+        timeoutSeconds: 600
+        updatePeriodSeconds: 1
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          app: che-plugin-registry
+          deploymentconfig: che-plugin-registry
+      spec:
+        containers:
+        - image: ${IMAGE}:${IMAGE_TAG}
+          imagePullPolicy: "${PULL_POLICY}"
+          name: che-plugin-registry
+          ports:
+          - containerPort: 8080
+          livenessProbe:
+            httpGet:
+              path: /plugins/
+              port: 8080
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            timeoutSeconds: 3
+          readinessProbe:
+            httpGet:
+              path: /plugins/
+              port: 8080
+            initialDelaySeconds: 3
+            periodSeconds: 10
+            timeoutSeconds: 3
+          resources:
+            limits:
+              memory: ${MEMORY_LIMIT}
+          envFrom:
+          - configMapRef:
+              name: che-plugin-registry
+              optional: true
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: che-plugin-registry
+  spec:
+    ports:
+      - protocol: TCP
+        port: 80
+        targetPort: 8080
+    selector:
+      deploymentconfig: che-plugin-registry
+- apiVersion: v1
+  kind: Route
+  metadata:
+    name: che-plugin-registry
+  spec:
+    to:
+      kind: Service
+      name: che-plugin-registry
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: che-plugin-registry
+  data:
+    CHE_SIDECAR_CONTAINERS_REGISTRY_URL: ${CHE_SIDECAR_CONTAINERS_REGISTRY_URL}
+    CHE_SIDECAR_CONTAINERS_REGISTRY_ORGANIZATION: ${CHE_SIDECAR_CONTAINERS_REGISTRY_ORGANIZATION}
+    CHE_SIDECAR_CONTAINERS_REGISTRY_TAG: ${CHE_SIDECAR_CONTAINERS_REGISTRY_TAG}
```

Comment:
@amisevsk I'm wondering about this cm, that will have empty values for plugin-registry. What impact will it have?
see the similar cm was created for devfile registry on dsaas-stg - `console/project/dsaas-preview/browse/config-maps/che-devfile-registry`